### ISSUE TITLE
File Upload: Limit concurrent requests

### ISSUE THREAD

**Issue by stefansidler** [oblique-bit/oblique#65]
Hi there

I would like to make a feature suggestions for the file upload component.

**Current situation:**
A user is allowed to drop/select as many files as he wants at the same time. This will be handled by the file upload component in a concurrent way which either leads to one massive request with all the files (singleRequest=false) or a lot of single requests in parallel (singleRequest=true).

**Problem:**
The load on our backend service can increase heavily if a user selects a lot of large files. This could even lead to out of memory exceptions in our case.
Also in the UI there is now a very long list of all the files which will get uploaded which is not very readable for a large list of files.

**Proposed solution:**
It would be nice to have a property like "maxConcurrentRequests" which would limit the amout of concurrent requests which will get fired in parallel when singleRequest is set to false.

**Example:**
- singleRequest = false
- maxConcurrentRequests = 5
- User selects 100 files
- There will always only be 5 requests concurrently. 95 files will get queued. After one request is finished, one file from the queue will get uploaded. And so on.

**Comment by c-saner** [comment]
Hello,

Thank you for your report 😃 

We'll look into this suggestion.

Kind regards

**Comment by punix81** [comment]
HI @josch83 , 

Thank you for your report 😃

We'll look into this suggestion.

Kind regards

**Comment by nina-egger** [comment]
Note: feature request tracked in OUI-3026

**Comment by nina-egger** [comment]
Will be implemented in an upcoming minor version on Oblique 11 or Oblique 12
